internal/service/streaminstance/resource_stream_instance.go

```diff
@@ -0,0 +1,213 @@
+package streaminstance
+
+import (
+	"context"
+	"errors"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ resource.ResourceWithConfigure = &streamInstanceRS{}
+var _ resource.ResourceWithImportState = &streamInstanceRS{}
+
+const streamInstanceName = "stream_instance"
+
+func Resource() resource.Resource {
+	return &streamInstanceRS{
+		RSCommon: config.RSCommon{
+			ResourceName: streamInstanceName,
+		},
+	}
+}
+
+type streamInstanceRS struct {
+	config.RSCommon
+}
+
+type TFStreamInstanceRSModel struct {
+	ID                types.String `tfsdk:"id"`
+	InstanceName      types.String `tfsdk:"instance_name"`
+	ProjectID         types.String `tfsdk:"project_id"`
+	DataProcessRegion types.Object `tfsdk:"data_process_region"`
+	Hostnames         types.List   `tfsdk:"hostnames"`
+}
+
+type TFInstanceProcessRegionSpecModel struct {
+	CloudProvider types.String `tfsdk:"cloud_provider"`
+	Region        types.String `tfsdk:"region"`
+}
+
+var ProcessRegionObjectType = types.ObjectType{AttrTypes: map[string]attr.Type{
+	"cloud_provider": types.StringType,
+	"region":         types.StringType,
+}}
+
+func (r *streamInstanceRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"instance_name": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"data_process_region": schema.SingleNestedAttribute{
```

Comment:
pending confirmation from PD if `cloud_provider` and `region` will be flattened into root level or not.

Reply:
who are we waiting for? is it Zuhair?

Reply:
Yes this is currently an open comment in the PD